builtin/logical/totp/path_code.go

```diff
@@ -94,6 +102,16 @@ func (b *backend) pathValidateCode(
 		return logical.ErrorResponse("an error occured while validating the code"), err
 	}
 
+	// Take the key skew, add two for behind and in front, and multiple that by
+	// the period to cover the full possibility of the validity of the key
+	err = b.usedCodes.Add(usedName, nil, time.Duration(
+		int64(time.Second)*
+			int64(key.Period)*
+			int64((2+key.Skew))))
```

Comment:
key.Period is not a duration, it's an integer number of seconds. If the period is 30 seconds and the skew is 1 period, that equation will give you time.Second * 32. What we want is time.Second * 90.